src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -218,7 +218,7 @@ private static function doRemove(array $files, bool $isRecursive): void
     public function chmod($files, int $mode, int $umask = 0000, bool $recursive = false)
     {
         foreach ($this->toIterable($files) as $file) {
-            if ((\PHP_VERSION_ID < 80000 || \is_int($mode)) && !self::box('chmod', $file, $mode & ~$umask)) {
```

Comment:
Why removing the php check?

Reply:
`$mode` is type hinted `int $mode`, so this condition can't be false.

Reply:
Although you are correct, I think we don't do this kind of change on LTS branches anymore.

Reply:
I wasn't sure also :smile: 

I rollbacked this and created a PR on 6.3 branch https://github.com/symfony/symfony/pull/49341